provision/README.md

```diff
@@ -1,43 +1,142 @@
-# PROVISION service
+# Provision service
 
-PROVISION service provides an HTTP API to interact with Mainflux.
+Provision service provides an HTTP API to interact with Mainflux. 
+Provision service is used to setup initial applications configuration i.e. things, channels, connections and certificates that will be required for the specific use case. 
+For example lets say you are using a Mainflux and connecting gateways to it. Gateways need an easy way to configure itself for communication with Mainflux (receiving and sending controls and data) and for that you will use bootstrap service but before gateway connects to the bootstrap service configuration needs to be created (things, channels, connections, bootstrap configuration and certificates if mtls is being used). Provision service should provide an easy way of provisioning your gateways. On a gateway there can be many services running and you may require more than one thing and channel to establish all the required means of representing a gateway in your application. Let's say that you are using an [Agent](https://github.com/mainflux/agent) and [Export](https://github.com/mainflux/export) service and you use mtls you will need to provision two things and certificates for those things for access to Mainflux, or you may want to create any number of things and channels for your application, this kind of setup we can call provision layout.
```

Comment:
I'm not sure that this is correct. `and you use mtls you will need to provision two things and certificates for those things for access to Mainflux`